internal/controller/components/reconciler.go

```diff
@@ -156,6 +185,44 @@ func Reconcile(
 			continue
 		}
 
+		if needsPullSecret {
+			providerSecret, skipCreation, err := makeProviderSecret(registryUsername, registryPassword, component.name, opts.Namespace, template)
+			if err != nil {
+				errMsg := fmt.Sprintf("failed to generate provider secret: %s", err)
+				updateComponentsStatus(statusAccumulator, component, template, errMsg)
+				errs = errors.Join(errs, errors.New(errMsg))
+				continue
+			}
+
+			if !skipCreation {
+				l.Info("Creating provider variable secret for " + component.name)
+				if err := mgmtClient.Create(ctx, providerSecret); client.IgnoreAlreadyExists(err) != nil {
+					errMsg := fmt.Sprintf("failed to patch provider secret %q for provider %s: %s", providerSecret.Name, component.name, err)
+					updateComponentsStatus(statusAccumulator, component, template, errMsg)
+					errs = errors.Join(errs, errors.New(errMsg))
+					continue
+				}
+			}
+
+			if component.targetNamespace != "" {
+				if err := kubeutil.CopySecret(
+					ctx,
+					mgmtClient,
+					mgmtClient,
```

Comment:
ditto: the first client is the source (always mgmt) while the second client is the target. In case of the management cluster, `rgnlClient == mgmtClient`.
```suggestion
					rgnlClient,
```